components/casemap/src/casemapper.rs

```diff
@@ -611,12 +611,13 @@ impl CaseMapper {
     /// );
     /// ```
     #[cfg(feature = "compiled_data")]
+    #[allow(clippy::new_ret_no_self)] // Intentional
```

Comment:
Thought: Rather than saying "Intentional" we should link to an issue or style guide chapter